index.bs

```diff
@@ -12403,158 +12294,157 @@ property is the String value "<code>forEach</code>".
-
-*   If the interface was declared with the [{{Global}}] [=extended attribute=],
-    then the property exists on the single object that [=implements=] the interface.
-*   Otherwise, the property exists solely on the interface’s [=interface prototype object=].
-
-If the interface has a [=value iterator=],
-then the [=function object=] is
-the value of the {{@@iterator}} property.
-
-<div algorithm="to invoke the values method of interfaces with a pair iterator">
-
-    If the interface has a [=pair iterator=],
-    then the method, when invoked, must behave as follows:
-
-    1.  Let |object| be the result of calling <a abstract-op>ToObject</a> on the <emu-val>this</emu-val> value.
-    1.  If |object| [=is a platform object=],
-        then [=perform a security check=], passing:
-        *   the platform object |object|,
-        *   the identifier "<code>entries</code>", and
-        *   the type "<code>method</code>".
-    1.  Let |interface| be the [=interface=]
-        on which the [=iterable declaration=] is declared.
-    1.  If |object| does not [=implement=] |interface|,
-        then [=ECMAScript/throw=] a {{ECMAScript/TypeError}}.
-    1.  Let |iterator| be a newly created [=default iterator object=]
-        for |interface| with |object| as its target and iterator kind "<code>value</code>".
-    1.  Return |iterator|.
-</div>
-
-<div algorithm="to invoke the values method of asynchronously iterable interfaces">
-
-    If the interface has an [=asynchronously iterable declaration=], then the method, when invoked,
-    must behave as follows:
-
-    1.  Let |object| be the result of calling [$ToObject$] on the <emu-val>this</emu-val> value.
-    1.  If |object| [=is a platform object=], then [=perform a security check=], passing:
-        *   the platform object |object|,
-        *   the identifier "<code>entries</code>", and
-        *   the type "<code>method</code>".
-    1.  Let |interface| be the [=interface=] on which the [=asynchronously iterable declaration=]
-        is declared.
-    1.  If |object| does not [=implement=] |interface|, then [=ECMAScript/throw=] a
-        {{ECMAScript/TypeError}}.
-    1.  Let |iterator| be a newly created [=default asynchronous iterator object=] for |interface|
-        with |object| as its [=default asynchronous iterator object/target=],
-        "<code>value</code>" as its [=default asynchronous iterator object/kind=], and
-        [=default asynchronous iterator object/is finished=] set to false.
-    1.  Run the [=asynchronous iterator initialization steps=] for |interface| with |object| and
-        |iterator|, if any such steps exist.
-    1.  Return |iterator|.
+<div algorithm>
+    To <dfn oldids="es-iterable-entries,es-iterable-keys,es-iterable-values">define the iterability</dfn>
+    of [=interface=] |definition| on |target|, given [=Realm=] |realm|, run the following steps:
+
+    1.  If |definition| has an [=indexed property getter=], then:
```

Comment:
I feel like the flow here could be enhanced a bit to make clear the implication and mutual exclusivity relations of the conditions, like:

1. If _definition_ has an indexed property getter, then:
   1. …
   1. If _definition_ has a value iterator, then …
1. Otherwise, if _definition_ has a pair iterator, then …
1. Otherwise, if _definition_ has an asynchronously iterable declaration, then …

You could also consider a switch-like syntax.